receiver/fluentforwardreceiver/receiver.go

```diff
@@ -97,6 +101,7 @@ func (r *fluentReceiver) Shutdown(context.Context) error {
 	if r.listener == nil {
 		return nil
 	}
+	time.Sleep(r.waitBeforeShutdownDuration)
```

Comment:
this doesn't seem like a good solution to me. It is possible more requests be made to the port while we wait.

Reply:
That's intended, this is to catch the case where (in my infra) AWS Firelens hasn't sent everything yet, so this adds a delay to ensure those logs are received

Reply:
Something is way off. I would build a test that continuously sends requests and counts them, and compares to successful logs received. On the first request refused, we stop the test and make sure we didn't drop any request.

The issue seems to come from the server.go file ; it's doing a few too many things with goroutines that can be problematic.